frontend/src/app/containers/ExperimentPage.js

```diff
@@ -649,19 +650,34 @@ export class ExperimentDetail extends React.Component {
     // Ignore subsequent clicks if already in progress
     if (isEnabling) { return; }
 
+    let installAddonPromise = Promise.resolve();
+    if (! this.props.hasAddon) {
+      let {installAddon, requireRestart, sendToGA, eventCategory} = this.props;
+
+      installAddonPromise = installAddon(
+        requireRestart,
+        sendToGA,
+        eventCategory);
+    }
+
     this.setState({
       isEnabling: true,
       isDisabling: false,
-      shouldShowTourDialog: true,
-      progressButtonWidth: evt.target.offsetWidth
+      shouldShowTourDialog: true
     });
 
-    enableExperiment(experiment);
-
-    sendToGA('event', {
-      eventCategory: 'ExperimentDetailsPage Interactions',
-      eventAction: 'Enable Experiment',
-      eventLabel: experiment.title
+    installAddonPromise.then(() => {
+      // XXX installAddonPromise shouldn't resolve until the add-on is ready
+      // to accept messages, instead of using a racy setTimeout here.
+      setTimeout(() => {
```

Comment:
We shouldn't do this setTimeout here, instead waiting on a promise that resolves on line 143 of frontend/src/app/lib/addon.js, where the console.log statement is below.